### PR TITLE
Avoid use of hardcoded providers

### PR DESCRIPTION
**Pull request by imonteroperez**
- Proposal from: https://github.com/jenkinsci/bouncycastle-api-plugin/pull/46#pullrequestreview-721264062
- Please @jtnord take a look and feel free to adapt whatever you consider :-)

### CODE REVIEW

File: src/main/java/jenkins/bouncycastle/api/PEMEncodable.java [hunk 1]
```diff
@@ -153,7 +174,7 @@ public static PEMEncodable decode(@Nonnull String pem, @Nullable final char[] pa
                 throw new IOException("Could not parse PEM, only key pairs, private keys, public keys and certificates are supported");
             }
 
-            JcaPEMKeyConverter kConv = new JcaPEMKeyConverter().setProvider("BC");
+            JcaPEMKeyConverter kConv = new JcaPEMKeyConverter().setProvider(BOUNCY_CASTLE_PROVIDER.getName());
```
Comment (jtnord) [inline src/main/java/jenkins/bouncycastle/api/PEMEncodable.java]: is `setProvider(Provider)` not available in both versions?

```suggestion
            JcaPEMKeyConverter kConv = new JcaPEMKeyConverter().setProvider(BOUNCY_CASTLE_PROVIDER);
```

File: src/main/java/jenkins/bouncycastle/api/PEMEncodable.java [hunk 2]
```diff
@@ -190,7 +211,7 @@ public static PEMEncodable decode(@Nonnull String pem, @Nullable final char[] pa
             } else if (object instanceof SubjectPublicKeyInfo) {
                 return new PEMEncodable(kConv.getPublicKey((SubjectPublicKeyInfo) object));
             } else if (object instanceof X509CertificateHolder) {
-                JcaX509CertificateConverter cConv = new JcaX509CertificateConverter().setProvider("BC");
+                JcaX509CertificateConverter cConv = new JcaX509CertificateConverter().setProvider(BOUNCY_CASTLE_PROVIDER.getName());
```
Comment (jtnord) [inline src/main/java/jenkins/bouncycastle/api/PEMEncodable.java]: is `setProvider(Provider)` not available in both versions?

```suggestion
            JcaX509CertificateConverter cConv = new JcaX509CertificateConverter().setProvider(BOUNCY_CASTLE_PROVIDER);
```

File: src/main/java/jenkins/bouncycastle/api/PEMEncodable.java (hunk 1 above)
Comment (imonteroperez) [reply src/main/java/jenkins/bouncycastle/api/PEMEncodable.java]: Yes, it is. Merging your suggestion now :-)

File: src/main/java/jenkins/bouncycastle/api/PEMEncodable.java [hunk 3]
```diff
@@ -190,7 +211,7 @@ public static PEMEncodable decode(@Nonnull String pem, @Nullable final char[] pa
             } else if (object instanceof SubjectPublicKeyInfo) {
                 return new PEMEncodable(kConv.getPublicKey((SubjectPublicKeyInfo) object));
             } else if (object instanceof X509CertificateHolder) {
-                JcaX509CertificateConverter cConv = new JcaX509CertificateConverter().setProvider("BC");
+            JcaX509CertificateConverter cConv = new JcaX509CertificateConverter().setProvider(BOUNCY_CASTLE_PROVIDER);
```
Comment (jtnord) [inline src/main/java/jenkins/bouncycastle/api/PEMEncodable.java]:
```suggestion
                JcaX509CertificateConverter cConv = new JcaX509CertificateConverter().setProvider(BOUNCY_CASTLE_PROVIDER);
```

**Pull request merged**